### PR TITLE
test databases populated with data to be used in testing

### PR DESCRIPTION
**Pull request by adamhind**

### CODE REVIEW

File: spec/setup_test_db.rb [hunk 1]
```diff
@@ -3,11 +3,37 @@
 def setup_test_database
   connection = PG.connect(dbname: 'ducks_bnb_test')
   connection.exec('TRUNCATE spaces CASCADE;')
+  connection.exec('TRUNCATE users CASCADE;')
+  connection.exec('TRUNCATE bookings CASCADE;')
 end
 
 def add_row_to_test_database
```
Comment (johnny-m-young) [inline spec/setup_test_db.rb]: Could we call this add_spaces_to_test_database, just so it's similar to the other methods?

File: spec/setup_test_db.rb [hunk 2]
```diff
@@ -3,11 +3,37 @@
...
   connection = PG.connect(dbname: 'ducks_bnb_test')
-  connection.exec("INSERT INTO spaces (space_name, description, price) VALUES ('Test Space #1', 'This is the first test space. Spacious.', 50.00);")
-  connection.exec("INSERT INTO spaces (space_name, description, price) VALUES ('Test Space #2', 'This is the second test space. Cosy.', 40.00);")
-  connection.exec("INSERT INTO spaces (space_name, description, price) VALUES ('Test Space #3', 'This is the third test space. Cheap.', 20.00);")
+  connection.exec("INSERT INTO spaces (space_name, description, price, available_from, available_to) VALUES 
+  ('Test Space #1', 'This is the first test space. Spacious.', '50.00', '01/01/2022', '01/30/2022');")
+  connection.exec("INSERT INTO spaces (space_name, description, price, available_from, available_to) VALUES 
+  ('Test Space #2', 'This is the second test space. Cosy.', '40.00', '01/01/2022', '01/30/2022');")
+  connection.exec("INSERT INTO spaces (space_name, description, price, available_from, available_to) VALUES 
+  ('Test Space #3', 'This is the third test space. Cheap.', '20.00', '02/01/2022', '02/28/2022');")
```
Comment (johnny-m-young) [inline spec/setup_test_db.rb]: I think the dates here need formatting in YYYY-MM-DD format

File: spec/setup_test_db.rb [hunk 3]
```diff
@@ -3,11 +3,37 @@
...
+
+def add_test_users_to_database
+  connection = PG.connect(dbname: 'ducks_bnb_test')
+  connection.exec("INSERT INTO users (first_name, surname, username, email, password) VALUES 
+  ('Test User 1', 'Number 1','TEST01','test1@test.com','Password1');")
+  connection.exec("INSERT INTO users (first_name, surname, username, email, password) VALUES 
+  ('Test User 2', 'Number 2','TEST02','test2@test.com','Password2');")
+  connection.exec("INSERT INTO users (first_name, surname, username, email, password) VALUES 
+  ('Test User 3', 'Number 3','TEST03','test3@test.com','Password3');")
+end
```
Comment (johnny-m-young) [inline spec/setup_test_db.rb]: Looks good!

File: spec/setup_test_db.rb [hunk 4]
```diff
@@ -3,11 +3,37 @@
...
+
+def add_test_bookings_to_database
+  connection = PG.connect(dbname: 'ducks_bnb_test')
+  connection.exec("INSERT INTO bookings (date_from, date_to, confirmed) VALUES 
+  ('01/01/2022', '01/02/2022','false');")
+  connection.exec("INSERT INTO bookings (date_from, date_to, confirmed) VALUES 
+  ('01/10/2022', '01/11/2022','false');")
+  connection.exec("INSERT INTO bookings (date_from, date_to, confirmed) VALUES 
+  ('01/01/2022', '01/02/2022','true');")
+end
```
Comment (johnny-m-young) [inline spec/setup_test_db.rb]: Do we need to add space_id in here? Not sure how the foreign key works.

File: spec/setup_test_db.rb (hunk 2 above)
Comment (johnny-m-young) [reply spec/setup_test_db.rb]: Do we need user_id in here? Again not sure how the FK works

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -25,6 +25,8 @@
   config.before(:each) do
     setup_test_database
     add_row_to_test_database
+    add_test_users_to_database
+    add_test_bookings_to_database
```
Comment (johnny-m-young) [inline spec/spec_helper.rb]: Looks good!

File: spec/setup_test_db.rb [hunk 6]
```diff
@@ -3,11 +3,37 @@
 def setup_test_database
   connection = PG.connect(dbname: 'ducks_bnb_test')
   connection.exec('TRUNCATE spaces CASCADE;')
+  connection.exec('TRUNCATE users CASCADE;')
+  connection.exec('TRUNCATE bookings CASCADE;')
```
Comment (johnny-m-young) [inline spec/setup_test_db.rb]: I think here, because we are truncating all tables, we don't need the cascade method on each.

**Review by johnny-m-young — COMMENTED** [review]
See comments. I have to update the Space class for the feature of filtering by dates, so don't worry too much about updating the tests for that because I will need to re-do them anyway.

File: spec/setup_test_db.rb (hunk 1 above)
Comment (adamhind) [reply spec/setup_test_db.rb]: Sorted :)

File: spec/setup_test_db.rb (hunk 2 above)
Comment (adamhind) [reply spec/setup_test_db.rb]: Completed :)

File: spec/setup_test_db.rb (hunk 4 above)
Comment (adamhind) [reply spec/setup_test_db.rb]: Completed

File: spec/setup_test_db.rb (hunk 6 above)
Comment (adamhind) [reply spec/setup_test_db.rb]: I have changed this to drop table now and left the cascade in as it was throwing an error if not.

**Review by johnny-m-young — COMMENTED** [review]
Looks good

**Comment by johnny-m-young** [comment]
Test

**Pull request merged**